b2share/modules/schemas/views.py

```diff
@@ -139,12 +183,128 @@ def get(self, community_schema):
         return community_schema
 
 
+class BlockSchemaListResource(ContentNegotiatedMethodView):
+    view_name = 'block_schema_list'
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(BlockSchemaListResource, self).__init__(
+            method_serializers={
+                'GET': {
+                    'application/json': schemas_list_to_json_serializer,
+                },
+                'POST': {
+                    'application/json': block_schema_to_json_serializer,
+                }
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def get(self):
+        """Get a list of all schemas."""
+        community_id = request.args['community_id']
+        schemas = BlockSchema.get_all_block_schemas(community_id=community_id)
+        return self.make_response(
+            schemas=schemas,
+            code=200)
+
+    def post(self):
+        """Create a new schema."""
+        if request.content_type != 'application/json':
+            abort(415)
+        data = request.get_json()
+        if data is None:
+            return abort(400)
+
+        schema = BlockSchema.create_block_schema(**data)
+        return self.make_response(
+            schema=schema,
+            code=201,
+        )
+
+
+class BlockSchemaResource(ContentNegotiatedMethodView):
+    view_name = 'block_schema_item'
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(BlockSchemaResource, self).__init__(
+            method_serializers={
+                'GET': {
+                    'application/json': block_schema_to_json_serializer,
+                },
+                'PATCH': {
+                    'application/json': block_schema_to_json_serializer,
+                }
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def get(self, schema_id):
+        """Get a schema."""
+        schema = BlockSchema.get_block_schema(schema_id)
+        self.check_etag(str(schema.updated))
+
+        return self.make_response(
+            schema=schema,
+            code=200
+        )
+
+    def patch(self, schema_id):
+        """Patch a schema."""
+        data = request.get_json(force=True)
+        if data is None:
+            abort(400)
+
+        block_schema = BlockSchema.get_block_schema(schema_id)
+        self.check_etag(str(block_schema.updated))
+
+        try:
+            if 'application/json' == request.content_type:
+                block_schema.update(data)
+            else:
+                block_schema = block_schema.patch(data)
+            db.session.commit()
+            return self.make_response(
+                schema=block_schema,
+                code=200
+            )
+        except (JsonPatchConflict):
+            abort(409)
+        except (JsonPatchException, InvalidJsonPatch):
```

Comment:
`InvalidJsonPatch` extends `JsonPatchException`. Thus there is no need to catch both of them.